### PR TITLE
Fixes #406 Use AJAX for toggle event in repository settings

### LINKED ISSUE #406: Use AJAX for toggle event in repository settings
### This is a
- [ ] Bug report
- [x] Feature request

### Description

There are few toggle events in repository's settings page. These include `Mail Service`, `Pull Request Status`, `Disable repository` among others. In the current implementation, we send a post request to their respective routes in the traditional way. Using AJAX and Bootstrap toggle would improve the UI as well as the UX

### PR DESCRIPTION
**Pull request by imujjwal96**
## Related Issue

Fixes #406 
## Motivation and Context

There are few toggle events in repository's settings page. These include Mail Service, Pull Request Status, Disable repository among others. In the current implementation, we send a post request to their respective routes in the traditional way. Using AJAX and Bootstrap toggle would improve the UI as well as the UX
## How Has This Been Tested?

Test at https://yaydoc7.herokuapp.com/
## Screenshots (if appropriate):
image

## Types of changes

- [ ] Bug fix 
- [x] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

**Comment by imujjwal96** [comment]
@pri22296 @sch00lb0y Please review

File: routes/repository.js [hunk 1]
```diff
@@ -87,26 +87,20 @@ router.post("/enable", authMiddleware.isLoggedIn, function(req, res, next) {
...
   .then(function () {
-    res.redirect('/' + req.body.repository + '/settings?status=mail_enabled_successful');
+    res.json({success: true});
   })
   .catch(function () {
-    next({
-      status: 500,
-      messages: 'Something went wrong'
-    });
+    res.json({success: false});
```
Comment (sch00lb0y) [inline routes/repository.js]: send `500` status only. handle the failure in the AJAX

Comment (imujjwal96) [inline routes/repository.js]: @sch00lb0y Updated

File: routes/repository.js [hunk 2]
```diff
@@ -87,26 +87,22 @@ router.post("/enable", authMiddleware.isLoggedIn, function(req, res, next) {
...
-    res.redirect('/' + req.body.repository + '/settings?status=mail_disabled_successful');
+    res.json({success: true});
   })
   .catch(function () {
-    next({
-      status: 500,
-      messages: 'Something went wrong'
-    });
+    res.status(500);
+    res.json({success: false});
```
Comment (sch00lb0y) [inline routes/repository.js]: I don't think so this is needed

Comment (imujjwal96) [inline routes/repository.js]: `res.json` ?

Comment (imujjwal96) [inline routes/repository.js]: When it was `next()`, a response was never received at the client side. Hence I had to make this change

Comment (sch00lb0y) [inline routes/repository.js]: are you sure? please check once.

Comment (imujjwal96) [inline routes/repository.js]: Oh! Just realised that the issue was caused due to a different reason which was fixesd. Checked it again and works with next. Thanks for the tip :)

**Pull request merged**